plugins/http_plugin/http_plugin.cpp

```diff
@@ -88,21 +88,21 @@ class http_plugin_impl : public std::enable_shared_from_this<http_plugin_impl> {
             auto next_ptr = std::make_shared<url_handler>(std::move(next));
             return [my=std::move(my), priority, next_ptr=std::move(next_ptr)]
                        ( detail::abstract_conn_ptr conn, string r, string b, url_response_callback then ) {
-               auto tracked_b = make_in_flight<string>(std::move(b), my->plugin_state);
+               auto tracked_b = make_in_flight(b, my->plugin_state);
```

Comment:
We should avoid this string copy.
#488 mentioned tracking bytes in flight in `beast_http_session` instead. That would allow for the removal of `make_in_flight` and the `in_flight` struct. The whole `in_flight` struct seems rather convoluted. It might have been needed at some point, but now it seems like overkill.

Can you try moving this into `beast_http_session` and see if it is cleaner. If that doesn't work see if a simplified `in_flight` is possible. Something as simple as 
`fc::make_scoped_exit([&](){atomic_in_flight -= count;});` might be possible.

Reply:
Kevin, thanks indeed my change was making an extra copy. I fixed it and reverted this code to its previous state. I'll look into the other changes you suggest.

Reply:
Hi Kevin, I don't think we can move the `bytes_in_flight` to the `beast_http_session`, because we want to keep track of the bytes in flight across all the sessions (see [here](https://github.com/AntelopeIO/leap/blob/03235ed3d2349c9b454212e78491883651a91b08/plugins/http_plugin/include/eosio/http_plugin/beast_http_listener.hpp#L113-L116)).
However, I think we can keep the count in `http_plugin_state`, and still remove the `in_flight` struct. Let me look into it.

Reply:
We increment `requests_in_flight` in `beast_http_session` constructor and decrement it in the destructor. Could you not decrement `bytes_in_flight` in the `beast_http_session` destructor?

Reply:
I don't think it would be a good idea, because the `beast_http_session` can remain as long as the socket is connected, serving multiple requests, and we want to decrement `bytes_in_flight` as soon as we have written the response to the socket.

Give me a little time to finish and test my changes, we can talk more then.

Reply:
@heifner I did remove the `make_in_flight` and the `in_flight` struct, and update the count of in_flight bytes directly. I think the code is simpler and easier to follow, and also it better keeps track of all in_flight bytes (including for error reports). It also removes the `shared_ptr` memory allocation and free to hold the payload.

Reply:
Nice!